packages/web3.js-plugin-fee-estimation/package.json

```diff
@@ -0,0 +1,45 @@
+{
+  "name": "@eth-optimism/web3.js-plugin-fee-estimation",
+  "version": "0.1.0",
+  "description": "A Web3.js plugin for doing OP-Chain gas estimation",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/ethereum-optimism/optimism.git",
+    "directory": "packages/web3js-plugin-fee-estimation"
+  },
+  "homepage": "https://optimism.io",
+  "type": "module",
+  "main": "dist/plugin.js",
+  "module": "dist/plugin.mjs",
+  "types": "src/plugin.ts",
```

Comment:
I'm the one you copied this shortcut from and I think it's fine to start but worth noting this generally should be building d.ts files rather than pointing at TS. It's fine to start here though just calling out that this is generally a shortcut and not a good pattern

Reply:
Added `dts: true` to [tsup.config.ts](https://github.com/ethereum-optimism/optimism/pull/6619/files#diff-1e8b4b032cda3302800f90e8377c75a685f1af5eb3cc605850d49944d9429404R13) and updated [package.json](https://github.com/ethereum-optimism/optimism/pull/6619/files#diff-4dd0e96c9dc5cb5cd01b42690052167fb26fef68952dee0fee7a730c0acdcce1R15) to: `"types": "dist/plugin.d.ts",`

Reply:
This is fine (especially for a package with 1 file) but you might not know this but this is also an antipattern. You generally want to avoid building types in the same pipeline as building your javascript (or building your types).  Typescript is such a performance problem at scale that you almost always want your typescript pipeline to be seperate from everything else.

As an example, in the https://github.com/latticexyz/mud repo I made their rebuilds go from 20 seconds to 200ms simply by removing a dts in their file.

Definitely fine here but a thing I try to tell everybody since it isn't common knowledge is to just be aware of this. If you can run your tests and build your javascript completely seperate from type generation and typechecking your builds will always be maximally performant.

As an example my [EVMts repo](https://github.com/evmts/evmts-monorepo/blob/main/package.json#L15) exposes a `build:dist` and `build:types` in every package and builds them seperately. The tests use vitest which uses esbuild under the hood (doesn't typecheck). This keeps everything peformant and the types build in parallel with the only thorn being I haven't optimized next.js example project to do the same thing yet

Reply:
I always tell folks to think of typescript as being a linter. For apps this works literally but for libraries the metaphor doesn't work as well since you do have to actually output types at the end of the day so it is a build step